### DISCUSSION TITLE
Support for Aggregate Signatures

### DISCUSSION THREAD

**Discussion by DrPeterVanNostrand** [filecoin-project/FIPs#840] [Filecoin Virtual Machine]
Currenty, FVM exclusively supports non-aggregate signatures via its `verify_signature` syscall, ideally FVM would support both aggregate and non-aggregate. BLS aggregate signatures allow multiple parties to sign multiple messages with a single constant-sized signature that is more computationally efficient to verify than verification of multiple individual signatures. As non-aggregate BLS is already supported in FVM, a new syscall could be added to allow verification of aggregate signatures without altering FVM's current interface use for verifying non-aggregate signatures.

FIP: 0079 (PR https://github.com/filecoin-project/FIPs/pull/841)

**Comment by AnomalRoil** [comment]
How does this relates to https://github.com/filecoin-project/FIPs/discussions/846 BTW?

From a gas cost perspective the two need to be well aligned so that the raw BLS pairing operations wouldn't be less costly than using the aggregate call.

**Reply by Stebalien** [reply]
In theory, #846 is more general purpose and _could_ replace this FIP. However, this FIP is:

1. More convenient (it does everything).
2. Likely more efficient for bulk verification as it avoids jumping back and forth between the actor and the FVM for each message being verifies.

I'd actually prefer to _just_ support the BLS12-381 primitives, but that's probably too slow (and also something that still needs a FIP/implementation).
